contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -406,4 +499,81 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
+  /// capabilities supported by the DON
+  /// @param isPublic True if the DON is public
+  function addDON(
+    bytes32[] calldata nodes,
+    CapabilityConfiguration[] calldata capabilityConfigurations,
+    bool isPublic
+  ) external onlyOwner {
+    uint32 id = s_donId;
+
+    s_dons[id].id = id;
+    s_dons[id].isPublic = isPublic;
+
+    for (uint256 i; i < nodes.length; ++i) {
+      bytes32 nodeP2PId = nodes[i];
+      if (s_dons[id].nodes.contains(nodeP2PId)) revert DuplicateDONNode(id, nodeP2PId);
+
+      s_dons[id].nodes.add(nodeP2PId);
+    }
+
+    for (uint256 i; i < capabilityConfigurations.length; ++i) {
+      CapabilityConfiguration calldata configuration = capabilityConfigurations[i];
+      bytes32 capabilityId = configuration.capabilityId;
+
+      if (!s_hashedCapabilityIds.contains(capabilityId)) revert CapabilityDoesNotExist(capabilityId);
+      if (s_deprecatedHashedCapabilityIds.contains(capabilityId)) revert CapabilityIsDeprecated(capabilityId);
+
+      if (s_dons[id].capabilityIds.contains(capabilityId)) revert DuplicateDONCapability(id, capabilityId);
+
+      for (uint256 j; j < nodes.length; ++j) {
+        bytes32 nodeP2PId = nodes[j];
+        if (!s_nodes[nodeP2PId].supportedCapabilityIds[s_nodes[nodeP2PId].configCount].contains(capabilityId))
+          revert NodeDoesNotSupportCapability(nodeP2PId, capabilityId);
+      }
+
+      s_dons[id].capabilityIds.add(capabilityId);
+      s_dons[id].capabilityConfigs[capabilityId] = configuration.config;
+    }
+
+    ++s_donId;
+
+    emit DONAdded(id, isPublic);
+  }
+
+  /// @notice Gets DON's data
+  /// @param donId The DON ID
+  /// @return uint32 The DON ID
+  /// @return bool True if the DON is public
+  /// @return bytes32[] The list of node P2P IDs that are in the DON
+  /// @return CapabilityConfiguration[] The list of capability configurations supported by the DON
+  function getDON(
+    uint32 donId
+  ) external view returns (uint32, bool, bytes32[] memory, CapabilityConfiguration[] memory) {
+    bytes32[] memory capabilityIds = s_dons[donId].capabilityIds.values();
+    CapabilityConfiguration[] memory capabilityConfigurations = new CapabilityConfiguration[](capabilityIds.length);
+
+    for (uint256 i; i < capabilityConfigurations.length; ++i) {
+      capabilityConfigurations[i] = CapabilityConfiguration({
+        capabilityId: capabilityIds[i],
+        config: s_dons[donId].capabilityConfigs[capabilityIds[i]]
+      });
+    }
+
+    return (s_dons[donId].id, s_dons[donId].isPublic, s_dons[donId].nodes.values(), capabilityConfigurations);
+  }
+
+  /// @notice Returns the DON specific configuration for a capability
+  /// @param donId The DON's ID
+  /// @param capabilityId The Capability ID
+  /// @return bytes The DON specific configuration for the capability
+  function getDONCapabilityConfig(uint32 donId, bytes32 capabilityId) external view returns (bytes memory) {
+    return s_dons[donId].capabilityConfigs[capabilityId];
```

Comment:
Should this also return the configuration contract's `getCapabilityConfiguration(donId)` as well?